### DISCUSSION TITLE
4-wire-version

### DISCUSSION THREAD

**Discussion by SchallertDK** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#247] [General]
Where in the "PlatformIO" do i find the "4-wire-version" as is mentioned in the files??

"3) Software
6-wire and 4-wire versions
DL and install Visual studio code: https://code.visualstudio.com/download
From within VSC, click on Extensions icon (ctrl+shift+X). Select PlatformIO IDE and install.
From the new platformio icon, open folder “6-wire-version” or “4-wire-version”."

**Comment by chunkysteveo** [comment]
It's in the "Code" folder of the project.

**Reply by SchallertDK** [reply]
Thanks a lot,
And where do i find the file "model.h" to choose my model (Miami):
"Choose your model in the file “model.h”. Only one model may be defined.
Double slashes ‘//’ means the line is commented out. To change model,
comment out PRE2021 by putting // in front of that line. Remove the ‘//’ from
your model."

I can find the "model.h" but i cant see where to choose my model?

**Reply by SchallertDK** [reply]
![image](https://user-images.githubusercontent.com/103250760/162801307-90ab753b-c4b6-4ee3-b1eb-fa4bbbdd79d8.png)

**Reply by visualapproach** [reply]
Hi! First figure out what model you have. Look on the pump for a number. Next find out how many wires goes between display and pump. If 4 you need to be able to determine which wire is what. (Using a DMM and/or looking at the pcb at the display or pump)
Report your findings here and we can point you in the right direction.
/visualapproach

**Comment by SchallertDK** [comment]
Hi.
Thanks for your help.
The pump is with 4 wire, and the cables are mounted as you see at the attached picture.
The pump is model #54123

Best regards

________________________________
Fra: visualapproach ***@***.***>
Sendt: mandag den 11. april 2022 21.26
Til: visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA ***@***.***>
Cc: SchallertDK ***@***.***>; Author ***@***.***>
Emne: Re: [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA] 4-wire-version (Discussion #247)


Hi! First figure out what model you have. Look on the pump for a number. Next find out how many wires goes between display and pump. If 4 you need to be able to determine which wire is what. (Using a DMM and/or looking at the pcb at the display or pump)
Report your findings here and we can point you in the right direction.
/visualapproach

—
Reply to this email directly, view it on GitHub<https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/discussions/247#discussioncomment-2547215>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AYTXWSFWFGVO5MDUSADS2G3VER4HTANCNFSM5TDIKSBQ>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

Jakob Schallert

**Reply by visualapproach** [reply]
I don't see any pictures but comment out the defined model:
//#define NO54138
And uncomment your model:
#define NO54123

Save file

Compile and upload.

Build and Upload filesystem

And again, find out what is 5V and GND! Do not gamble here.

**Reply by SchallertDK** [reply]
![image](https://user-images.githubusercontent.com/103250760/162824823-330073fe-a41d-429f-b967-d978b964c994.png)
![image](https://user-images.githubusercontent.com/103250760/162824897-8da178e3-6e89-4f05-9894-e0a7c37ff54f.png)

How does it work to control from an app? :)

**Reply by SchallertDK** [reply]
![image](https://user-images.githubusercontent.com/103250760/162825351-ef71961e-c02b-4d08-892a-9425fa8b715a.png)
Like this right?

**Reply by visualapproach** [reply]
Yeah, that's it. And they labeled the wires. Very kind of them.
There is no app. The device has a web server and you just browse to the ip from safari or whatever browser you fancy. Like http://192.168.Xx.yy
(After you entered the Wi-Fi credentials in the access point at first run)

**Reply by SchallertDK** [reply]
Thanks again :)
"(After you entered the Wi-Fi credentials in the access point at first run)" is it something who is coming up after loading to the device?

**Reply by visualapproach** [reply]
Yes.

**Reply by SchallertDK** [reply]
As my picture shows, i have GND - TXD - RXD - 5V at my pump print, hov should they be mounted at your print? (H3 and H4)
5V is in the square and GND is under that (comparing to the 6 wire  setup?) but i dont see what D3 and D2, and D6 and D7 should TXD and RXD
![image](https://user-images.githubusercontent.com/103250760/162919971-acab60be-6072-4b08-959c-8712f6b60ff0.png)

**Reply by SchallertDK** [reply]
![image](https://user-images.githubusercontent.com/103250760/162918892-55746e3f-7378-43ea-8756-b1f6b9a7f52c.png)

**Reply by visualapproach** [reply]
I'll reply tonight. Busy now

**Reply by SchallertDK** [reply]
No stress, answer when you have the time for it ;)
Your print is first arriving 19.04.22 :)

**Comment by chunkysteveo** [comment]
![20220111_141315](https://user-images.githubusercontent.com/14663002/162923158-badcadbe-0a9f-448e-87a4-201a2edfaf89.jpg)

This image should help - your display goes to the bottom set of 4. Your pump unit 'CIO' goes to the top set of 4. 5v is the top pin on both. GND is the second to top pin on both, next to the 5V. For my pump the white with solid red dash went to pin 3 (counting from top to botto), and the white with red X 'cross' went to the 4th pin. Replicate the same on both sets of 4 connectors. If you don't get any control of the pump when testing the two wires may be the wrong way round. You won't damage anything if they are, just don't get the 5V and GND messed up.

You will need to twist them as shown, as the connector is 5v-data-data-GND, but the PCB is 5v-GND-data-data.

Good luck!!

Hope that helps.

**Reply by SchallertDK** [reply]
Thank you :) This should help a lot :)

**Reply by visualapproach** [reply]
What @chunkysteveo said

**Reply by SchallertDK** [reply]
What is the code to the AcessPoint "Lay-Z-Spa Module"?

**Reply by visualapproach** [reply]
> What is the code to the AcessPoint "Lay-Z-Spa Module"?

Chapter 6 in the build instructions

**Reply by SchallertDK** [reply]
Does my computer need to have wifi to do the setup? I have cable, not wifi to my gamer computer where i am sitting and trying with your help to get i working :)

**Reply by visualapproach** [reply]
Your computer doesn't need Wi-Fi, but if you want to control the device it needs Wi-Fi. If you want to use your computer to update the esp over the air, or control it from the computer, they should be on the same network.

**Reply by SchallertDK** [reply]
After telling witch wifi it should work under and typing the password to the wifi, i cant find the module anymore? i have tried the ip adresses it tells when i open "serial monitor", and i can see it is trying to coonect, but it says "404: File Not Found"

**Reply by visualapproach** [reply]
Try consulting the build instructions again

**Reply by SchallertDK** [reply]
It says:
�␎$Failed to load settings.txt
Failed to read cmdq.txt       
WiFi > using WiFi configuration with SSID "Wifi34_2G"
WiFi > Trying to connect ....
.
.
.
.
.
WiFi > Connected. 
 SSID: "Wifi34_2G"
 IP: "192.168.87.122"
OTA > ready
HTTP > server started
WebSocket > server started
Failed to read mqtt.json. Using defaults.
MQTT > Buffer size successfully increased
End of setup()

**Reply by visualapproach** [reply]
Awesome!

**Reply by SchallertDK** [reply]
Is it okay?
How do i connect and see the screen where i can setup up the features to the pump?

**Reply by SchallertDK** [reply]
After the guide i'm at standing at step "3. Software":
I have found the module with it's own wifi acess point, and choosed with network it should choose + password to the network.
After that i cant find the module (ESP8266)?
The module is not blinking blue anymore but is lightning constant blue.

But after the guide, next step is "4. Web interface" where i should find this site, typing the chossen IP adress. But i only get an "404: File Not Found" ?
I can't hit the "/index.html" as showed in the guide, and buy typing my ip adress i should end up with this site, right?

Can i do everything over again? How do i delete the file in the ESP to start over? I have tried to upload file again, but the wifi i not switching back to the first default acess point, it stays on my wifi i have configurated to it.

![image](https://user-images.githubusercontent.com/103250760/163114025-20e2f546-6d4a-4dfa-b7bc-aea2a0affe92.png)
![image](https://user-images.githubusercontent.com/103250760/163112464-a5d34a96-8dc3-42d1-a6a5-44db1361f60b.png)

**Reply by chunkysteveo** [reply]
Sounds like you have not done the step in YELLOW to upload the file system data, or that step failed? 

You need to click the two tasks to BUILD the filesystem, and then UPLOAD the filesystem. (Top right box on screenshot) 
Project Tasks->Platform->Build Filesystem Image - click that, and then click Project Tasks->Platform->Upload Filesystem Image 
![image](https://user-images.githubusercontent.com/14663002/163127694-60649381-71a2-4bcc-a43e-e25985d38a22.png)

**Reply by SchallertDK** [reply]
Ahhh... Thank you.
I see this stuff is not for amatures as me. There are so many things i dont now where to find/or where to look.
So thank you for your help and patience

**Reply by SchallertDK** [reply]
It is stil showing "404: File Not Found" when i'm typing my ip adress?

![image](https://user-images.githubusercontent.com/103250760/163135267-ad0ec572-a84c-460b-ac47-3e63e7e366a3.png)

**Reply by chunkysteveo** [reply]
Correct - this is not a simple setup. There's only so much advise and direction we can give, but the instructions are fairly well documented that you should get there eventually!

**Reply by chunkysteveo** [reply]
Have you done the "Build File System Image" and "Upload File System Image" ?

**Reply by SchallertDK** [reply]
I got it as i had to click at "build filesystem image" and then upload to the ESP again, i didn't get it as your describe know (sorry)
But it fail uploading the filesystem image? I have tried also the OTA one.
![image](https://user-images.githubusercontent.com/103250760/163141399-d20e6fbb-6125-42df-b928-e35a7e2130b3.png)


Is it because i have choosen the OTA upload protocol?
![image](https://user-images.githubusercontent.com/103250760/163141333-a4bc946e-cb11-4631-96ba-9061b23cc6a2.png)

**Reply by SchallertDK** [reply]
I switched back to "esptool" and tried everything one more time, and everything has been uploaded with "succes"
I changed back to upload protocol espota, and now i can connect.
Thank you once again

**Reply by chunkysteveo** [reply]
Good news!

**Reply by SchallertDK** [reply]
How do the SPA config - planning works?
An example:
I have made 3 commands, but im not sure if it is right?
I want the pumpe and heater to start every fridag at 10 am and i want the setpoint to be 38 degress.

i have choosen "set target", "Set heater" and "set filter pump" - Is that right? Or does the pump start automatic just like normal when i only set the heater on the pump display?

![image](https://user-images.githubusercontent.com/103250760/163146022-e9e91f46-d711-4777-ba41-5607dac1d8e9.png)

**Reply by SchallertDK** [reply]
And next, what is the MQTT Config? And if i have to use it, how do i set it up? :)
![image](https://user-images.githubusercontent.com/103250760/163147491-b60b2cc6-212d-49d5-9f61-6e8deb5353aa.png)

**Reply by visualapproach** [reply]
If you turn on the heater, the pump will start automatically.
If you turn off the pump the heater will also turn off. 
The value must be 0 for off, and 1 for on. Not 38.

**Reply by visualapproach** [reply]
> And next, what is the MQTT Config? And if i have to use it, how do i set it up? :) ![image](https://user-images.githubusercontent.com/103250760/163147491-b60b2cc6-212d-49d5-9f61-6e8deb5353aa.png)

As stated in the build instructions, it's for advanced users, and if you don't know what it is you shouldn't worry about it.

**Reply by SchallertDK** [reply]
Thanks a lot :)

**Reply by SchallertDK** [reply]
Hi again,
Don't know what happens, but after last time i mountet today everything, but cant reach the module anymore?
I have unpluged it from the spa and connected it to my pc again, and it cant find it?
The light on the print i lightning blue, not flashing. So it is like it is connected to the wifi (?)
![image](https://user-images.githubusercontent.com/103250760/164519612-daef4a9d-eec9-4530-b49c-5b6e72fbcb0e.png)

**Reply by visualapproach** [reply]
Firmware version?
Pump version?

**Reply by SchallertDK** [reply]
I dont understand what you mean?
Is it possible to delete everything on the module and start over?

**Reply by visualapproach** [reply]
I asked what version of the program you have uploaded to your esp. Yes you can erase and upload new stuff. The reason I asked is because a recent version had to be mounted in the pump to show signs of life. I suggest you download the latest from github and upload to esp.
The reason I asked about pump model is because there is several versions out there and they behave differently. In no version the blue LED has anything to do with Wi-Fi.

**Reply by visualapproach** [reply]
FAQ: https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/discussions/46#discussion-64636

**Reply by SchallertDK** [reply]
Sorry for being not so proff at this stuff.
Where do i see my firmware version?
How do i erase what is on the esp?
My pump is af Model #54123

**Reply by visualapproach** [reply]
No worries, going outside of one's convenient space is healthy. I just don't wanna answer the same questions over and over again so I wrote a FAQ and instructions to avoid that, and that's why I rather point you to those instead of giving straight answers.
Now, the firmware version is a date, defined in config.h file, at least in the 6 wire version. And there is also a github tag like v3.2.1 on the homepage. But never mind. You have a 4 wire if I'm right, which shouldn't have the problem I spoke about. You can but don't need to erase the flash on the esp. just upload whatever new code you want and it will overwrite the old stuff.

**Reply by SchallertDK** [reply]
Now it works again, and hope it will continue that way :) 
Thanks again
But i can see, i need to find out how this homeassistent or MQTT works, so i can control the pumpe from everywhere, and not only over the same wifi

**Reply by SchallertDK** [reply]
Why is the module resetting my settings all the time?
My commands under "SPA Config" is the only thing it is remembering, but cost price, Chlorine and Filter change interval is switching back to "standart" all the time, and the same for Temperatyre setpoint and the "timer" for chlorine and filter change on the "Home" page
I had made a command to start the heater and set to 38 degress today at 8 am, but when i look at the webpage i couldn't see if the pump and heater was activated? The "Heater" bottom was black, "Pump" bottom was white and "take control" was switched off again (white)

So i switched to "Take control" and activated it by my self. (Shouldn't i be able to see at the webpage what is running and wat not is running?)

**Reply by visualapproach** [reply]
The configs should be saved when you press save. Timers and uptime etc is saved every few hours. You can see what's saved by first looking up the filedirectory by visiting IP/dir/ and if you want to see what's in a file go to IP/filename I.e. 192.168.0.10/settings.txt

**Reply by SchallertDK** [reply]
It is also saved at the beginning, but then after time when i'm entering the IP adress again to see, then it is reset again?

**Reply by visualapproach** [reply]
No shouldn't do

**Reply by chunkysteveo** [reply]
Hmmm, there may be an issue with the 4-wire version - i've noticed my chlorine and filter timers never seem to "stick" between reboots, and my cost per kWH would always reset back - like the settings were never being saved.

Never stopped to diagnose or troubleshoot it, but there may be an issue there with retaining or saving settings.

**Reply by visualapproach** [reply]
Ok I'll look at it

**Reply by chunkysteveo** [reply]
OK did some testing to be sure - on a 4 wire. 

I click the red "Reset" buttons to reset the Chlorine and Filter changed days to 0.0 days. I set the Spa Config Price per kWh to 0.2 and save. And I then reset the ESP using the red Reset ESP in the menu.

Here's the settings.txt file after saving, and then after a reboot - the savetime shows it's been overwritten on reboot maybe? But you can see the second settings.txt json object shows the values have been "reset"?

{
  "CLTIME": 1650626055,
  "FTIME": 1650626059,
  "UPTIME": 102,
  "PUMPTIME": 0,
  "HEATINGTIME": 0,
  "AIRTIME": 0,
  "JETTIME": 0,
  "TIMEZONE": 0,
  "PRICE": 0.200000003,
  "FINT": 30,
  "CLINT": 7,
  "AUDIO": 1,
  "SAVETIME": "2022-04-22 11:14:45"
}

{
  "CLTIME": 0,
  "FTIME": 0,
  "UPTIME": 86,
  "PUMPTIME": 0,
  "HEATINGTIME": 0,
  "AIRTIME": 0,
  "JETTIME": 0,
  "TIMEZONE": 0,
  "PRICE": 1,
  "FINT": 30,
  "CLINT": 14,
  "AUDIO": 1,
  "SAVETIME": "2022-04-22 11:16:39"
}

**Comment by visualapproach** [comment]
There was a bonehead mistake in the setup. Variables were initialized after settings were loaded. Fix uploaded in the development version now.

**Reply by SchallertDK** [reply]
Nice, and where do i find this "development version"?

**Reply by SchallertDK** [reply]
I think i found it, but how do i upload it to the "Visual Studio Code"?

https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/commit/4c27f77fa5d8e0f311321602740765d71341dcae

**Reply by SchallertDK** [reply]
Or to the esp?

**Reply by SchallertDK** [reply]
I have changes every 16 files to those you have uploaded in the development folder, to the "visual studio code", and then uploaded to esp.

**Reply by SchallertDK** [reply]
But when i refresh the website, it still goes back to factory settings at the main page?
![image](https://user-images.githubusercontent.com/103250760/164780341-cd2d29f3-9609-4b69-8884-58f76aabd1ed.png)

**Reply by SchallertDK** [reply]
And this is how i normal set it up:
![image](https://user-images.githubusercontent.com/103250760/164780471-48e5c978-d09d-4317-b436-7b0dcf3296d2.png)

**Reply by SchallertDK** [reply]
And the pump didn't stop at 8 PM as i had made a weekly command for? I had to turn it of by my self. (The command is gone know and replaced at the new command to next week)
![image](https://user-images.githubusercontent.com/103250760/164781069-4f991627-7baf-49b5-b439-e09f6d35700a.png)

**Reply by visualapproach** [reply]
You don't upload anything to VSC. Just download the whole shabang and open the new folder from VCS. Compile, upload, build filesystem, upload. I see you haven't updated the html file. Not that it matters, but still.
The states is not remembered after a reboot. That function is not implemented on the 4-wire system yet. But the config should be saved when you press save.

**Reply by visualapproach** [reply]
And by the way, the 23:rd of april hasn't happened yet, so why  did you expect it to do anything? 
Edit: oh it was set to 22/4 and rolled over to 29/4.. don't know
And why do you have multiple commands at the same time?

**Reply by SchallertDK** [reply]
I have build multiple commands because i thougt the pump needed the info for every time?
*Temperature setpoint (same time as heater start(?))
*Heater activation
*Pump stop time

**Reply by visualapproach** [reply]
But you set the same target 3 times. And heat on twice. Probably not going to hurt anything but it looked strange

**Reply by SchallertDK** [reply]
After update hole "shabang" to the ESP the program resets and stops the pump almost all the time?
And know the commands also is gone, chlorine, price and filter change is also default everytime?

I dont think i understand the commands right?
I have a command rutine for friday, saturday and Sunday
And i thougt i had to set the commands for each off those days?
Also at startup at the morning:
Heat temp. + Heater on 
Shutdown at the evening:
Pump of

And this i set to repeat every week (fri, sat, sun.)

**Reply by SchallertDK** [reply]
Since it's not working, i maybe had made something wrong?
Today i try downloading the development code
https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/tree/development

But there is no PlatformIO file to open, in this "code"?

**Reply by visualapproach** [reply]
> But there is no PlatformIO file to open, in this "code"?

No, the "platformio.ini" is in the 6-wire / 4-wire folder, just as when you did this the first time.

This should do what you want. Take control must be active ofc.
Set target:38 2022-04-29 08:00:00 interval: 604800s (this isn't really necessary if you don't change the target temperature manually or reset the esp)
Set heat:1 2022-04-29 08:00:00 interval: 604800s
Set pump:0 2022-04-29 20:00:00 interval: 604800s
Set heat:1 2022-04-30 08:00:00 interval: 604800s
Set pump:0 2022-04-30 20:00:00 interval: 604800s
Set heat:1 2022-05-01 08:00:00 interval: 604800s
Set pump:0 2022-05-01 20:00:00 interval: 604800s

Check if the ESP restarts unintentionally. You can see last restart on the SPA Config page.

**Reply by SchallertDK** [reply]
I think i'm not understanding how the update to your fixed files is working?
Can you send me the link, so i'm sure it is the right code i'm downloading?
It is maybe not the one i linked to, that i should use?

The last reboot was 13.00 (1 pm) - And it wasn't me. So the spa was off when i went home at 3 pm.
![image](https://user-images.githubusercontent.com/103250760/164988726-c5e28441-a777-478c-9f82-7b629103b8a5.png)

**Reply by visualapproach** [reply]
build instructions ch 3:
![image](https://user-images.githubusercontent.com/22987250/164988911-f840712e-1cc5-47cb-83b1-9fd6cc29b015.png)

OR upload files via web ui menu "upload files". It's the files in the "data" directory. You may wanna go this route to save previous settings. Only upload .html and .js files.

But you have a problem with the ESP resetting. Bad Wi-Fi or maybe bad power supply? (some 4-wire pump appear to have problem deliver enough current according to users)

**Reply by SchallertDK** [reply]
This is also the way i'm doing. But my question was if it maybe is the wrong files i'm using/uploading?
You told you had found somethings and fixed them, but the only files i can se recently is changes at the 4 pin files, is not in the master directory, but in the development.
Maybe many at the faults is because of the maybe missing constant 5v power as you says can happen?

**Comment by visualapproach** [comment]
![image](https://user-images.githubusercontent.com/22987250/164990662-b82066a4-52e0-4dda-a0a9-f8da86579df0.png)

extract all files. Start VCS.  From platformio open project, browse to 4-wire.

**Reply by SchallertDK** [reply]
I have tried that, but PlatformIO says it cant open because there is no PlatformIO file, as there is in the "master" zip file. Because this I can open with no problems

**Comment by visualapproach** [comment]
Yes there is!
![image](https://user-images.githubusercontent.com/22987250/164992177-301d2eb5-5a53-4432-9ea4-555352ee0d38.png)

**Reply by SchallertDK** [reply]
Oh... Now i found it.
Thank you again.
When i'm uploading filesystem image OTA, it ends withs this error:
![image](https://user-images.githubusercontent.com/103250760/164992751-96b4d931-8c12-44b7-8918-e02dc5368ed8.png)

But the build and upload after that is running with succes? So is this filesystem image error god or bad?

**Reply by visualapproach** [reply]
Use the usual file system upload. Not the OTA. It will work over the air anyway if you configured the ini file

**Reply by SchallertDK** [reply]
Just because i have to unmount it from the spa outside every time then 😅

**Reply by visualapproach** [reply]
No I said it should work over the air. Just use the right tool. Not the one that says OTA.

**Reply by SchallertDK** [reply]
What tool is it you are regarding to? 😅

**Reply by visualapproach** [reply]
> Use the usual file system upload. Not the OTA. 

I meant this, but maybe it doesn't work either. So upload via web ui.
![image](https://user-images.githubusercontent.com/22987250/164995283-46595d7f-1baf-4c90-9fb1-93e5c45c1d91.png)

**Reply by SchallertDK** [reply]
Sorry again, but where do i find that and what is web ui?

**Reply by visualapproach** [reply]
web ui = web user interface = read build instructions chapter 4

**Reply by SchallertDK** [reply]
Yeah okay, Now everything went smooth with no errors.
So the issue with the factory settings at the ESP Module is the 5 volt power who maybe isn't constant?

**Reply by SchallertDK** [reply]
Hi agian,
Again i lost the connection to my module. Why do this happen?
I have to unplug the module from the spa, and reinstall everything and get af new ip adress.
This is second time

**Reply by chunkysteveo** [reply]
Use a static IP from your router maybe?

**Reply by SchallertDK** [reply]
I really dont know how and what to do.
In the index - network config i choose static ip and typed the ip, but the nothing worked
I tried to start all over again with downloading af new code and installed it.
Everything seems good, but at the  computer where the esp is conected and got the setup, i can connect the ip adress, but from my phone on the same network i cant? I know i has done something wrong, but cant figure out what it is?

**Reply by visualapproach** [reply]
You shouldn't need to reinstall it just because the IP changed. If that's the problem...
We cannot help with the sparse info you gave. You need to give all the details about your problem and exactly what you do if anyone should be able to say anything useful.
Things we don't know: 
Where is your layzspa? 1 km from the router behind a reinforced wall?
What router do you have?
What IP did you enter? Screenshots?
What's the IP of the router and what's the network settings? Reserved addresses possible?
What phone and IP?
etc etc

The more info you give the more likely someone is to be able to help you.
